packages/core/src/domain/configuration/configuration.ts

```diff
@@ -107,6 +107,9 @@ export interface Configuration extends TransportConfiguration {
   batchMessagesLimit: number
   messageBytesLimit: number
 }
+function isDatadogSite(site: string) {
+  return /(datadog|ddog|datad0g|dd0g)/.test(site)
```

Comment:
Maybe check that it's followed by a TLD `/(datadog|ddog|datad0g|dd0g).(com|eu)/` otherwise `datadog.example.org` would be true for example.